### PR TITLE
storage: Support for creating biosboot and efi partitions

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

File: pkg/storaged/block/format-dialog.jsx [hunk 1]
```diff
@@ -185,6 +185,12 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
     add_fsys("ext4", { value: "ext4", title: "EXT4" });
     add_fsys("vfat", { value: "vfat", title: "VFAT" });
     add_fsys("ntfs", { value: "ntfs", title: "NTFS" });
+    if (client.in_anaconda_mode()) {
+        if (block_ptable && block_ptable.Type == "gpt")
+            add_fsys(true, { value: "biosboot", title: "BIOS boot partition" });
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:190]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19826-20240109-133522-cb1438ac-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 2]
```diff
@@ -185,6 +185,12 @@
...
+        if (block_ptable)
+            add_fsys(true, { value: "efi", title: "EFI system partition" });
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:192]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19826-20240109-133522-cb1438ac-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 3]
```diff
@@ -406,6 +412,8 @@
                 } else {
                     dlg.update_actions({ Variants: [action_variant], Title: action_title });
                 }
+                if (vals.type == "efi" && !vals.mount_point)
+                    dlg.set_values({ mount_point: "/boot/efi" });
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:416]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19826-20240109-133522-cb1438ac-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 4]
```diff
@@ -416,6 +424,18 @@
             disable_on_error: usage.Teardown,
             action: function (vals) {
                 const mount_now = vals.variant != "nomount";
+                let type = vals.type;
+                let partition_type = null;
+
+                if (type == "efi") {
+                    type = "vfat";
+                    partition_type = block_ptable.Type == "dos" ? "0xEF" : "c12a7328-f81f-11d2-ba4b-00a0c93ec93b";
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:432]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19826-20240109-133522-cb1438ac-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 5]
```diff
@@ -416,6 +424,18 @@
...
+                }
+
+                if (type == "biosboot") {
+                    type = "empty";
+                    partition_type = "21686148-6449-6e6f-744e-656564454649";
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:437]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19826-20240109-133522-cb1438ac-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 6]
```diff
@@ -532,11 +552,19 @@
...
                     } else {
-                        return block.Format(vals.type, options);
+                        return block.Format(type, options);
+                    }
+                }
+
+                async function set_partition_type(new_path) {
+                    if (partition_type) {
+                        const block_part = await client.wait_for(() => client.blocks_part[new_path]);
+                        await block_part.SetType(partition_type, {});
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:565]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19826-20240109-133522-cb1438ac-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/partitions/partition.jsx [hunk 7]
```diff
@@ -118,6 +141,62 @@ const PartitionCard = ({ card, block, unused_space_warning, resize_info }) => {
...
+                    return _("Type must be of the form NNNNNNNN-NNNN-NNNN-NNNN-NNNNNNNNNNNN.");
+            }
+        }
+
+        dialog_open({
+            Title: cockpit.format(_("Set partition type of $0"), block_name(block)),
+            Fields: [
+                SelectOne("type", _("Type"),
+                          {
+                              value: (type_names[type] ? type : "custom"),
```
Comment (cockpituous) [inline pkg/storaged/partitions/partition.jsx:177]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19826-20240109-133522-cb1438ac-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 8]
```diff
@@ -416,6 +425,18 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
             disable_on_error: usage.Teardown,
             action: function (vals) {
                 const mount_now = vals.variant != "nomount";
+                let type = vals.type;
+                let partition_type = "";
+
+                if (type == "efi") {
+                    type = "vfat";
+                    partition_type = block_ptable.Type == "dos" ? "0xEF" : "c12a7328-f81f-11d2-ba4b-00a0c93ec93b";
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:433]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19826-20240112-122255-20b1af54-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 9]
```diff
@@ -517,13 +538,13 @@
 
                 function format() {
                     if (create_partition) {
-                        if (vals.type == "dos-extended")
+                        if (type == "dos-extended")
                             return block_ptable.CreatePartition(start, vals.size, "0x05", "", { });
-                        else if (vals.type == "empty")
-                            return block_ptable.CreatePartition(start, vals.size, "", "", { });
+                        else if (type == "empty")
+                            return block_ptable.CreatePartition(start, vals.size, partition_type, "", { });
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:544]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19826-20240112-122255-20b1af54-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 8 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:433]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19826-20240112-132957-b00e3528-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 10]
```diff
@@ -517,13 +538,13 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
...
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:544]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19826-20240112-132957-b00e3528-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by KKoukiou — APPROVED** [review]

File: pkg/storaged/block/format-dialog.jsx (hunk 8 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:433]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19826-20240115-072533-b00e3528-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 10 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:544]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19826-20240115-072533-b00e3528-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Pull request merged**